src/main/java/com/google/devtools/build/lib/starlarkbuildapi/BuildConfigurationApi.java

```diff
@@ -75,6 +75,16 @@ public interface BuildConfigurationApi extends StarlarkValue {
               + " function.")
   boolean isCodeCoverageEnabled();
 
+  @StarlarkMethod(
+      name = "short_id",
+      structField = true,
+      doc = """
+            A short identifier for this configuration understood by <code>bazel config</code>.
+
+            <p>Use of this field should generally be avoided as it hurts cacheability.
```

Comment:
Can we tweak this phrasing?

The full API space is of course hard to follow and users struggle knowing which calls are recommended, deprecated, overlapping, etc.

This could be read as adding a new call that we immediately tell people not to use. Can we adjust this to somehow compare its value & costs to help users feel best informed in whether to use it or not?